pandas/core/indexes/base.py

```diff
@@ -958,6 +958,10 @@ def take(
         if kwargs:
             nv.validate_take((), kwargs)
         indices = ensure_platform_int(indices)
+
+        if not indices.ndim == 1:
```

Comment:
I don't think this will work.
For the case, `df.take(1)`, here indices in L960 would be `np.array(1)`. So what you're essentially checking is `np.array(1).ndim`, which would be 0.
It would be best to check on your local if this test / change is working as expected. You can see how to run the tests locally [here](https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#running-the-test-suite).I can also see related test failures [here](https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=64842&view=logs&jobId=404760ec-14d3-5d48-e580-13034792878f&j=404760ec-14d3-5d48-e580-13034792878f&t=f81e4cc8-d61a-5fb8-36be-36768e5c561a).

Reply:
@debnathshoham I wanted to confirm one thing. Our goal is to raise a TypeError when a scalar parameter is passed in `df.take()`. right?

Reply:
That's correct.

Reply:
@debnathshoham Going by my implementation, `df.take(1)` has `ndim=0` (as you stated). Thus a TypeError is being raised.
```py
>>> import pandas as pd
>>> df = pd.DataFrame({'a': [1, 2, 3]})
>>> df.take(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/arijitroy/Projects/pandas/pandas/core/generic.py", line 3618, in take
    new_data = self._mgr.take(
  File "/Users/arijitroy/Projects/pandas/pandas/core/internals/managers.py", line 848, in take
    new_labels = self.axes[axis].take(indexer)
  File "/Users/arijitroy/Projects/pandas/pandas/core/indexes/range.py", line 437, in take
    return self._int64index.take(
  File "/Users/arijitroy/Projects/pandas/pandas/core/indexes/base.py", line 963, in take
    raise TypeError("Expected indices to be 1 dimensional")
TypeError: Expected indices to be 1 dimensional
>>> 
```

Reply:
I don't think you've push this commit. In the latest commit 283d56b2ce0fff77d6e59dde5c5f5c0ffea44384 you're  comparing with 1. 
Push this change, and also update the test (comment below)